src/waveform/renderers/waveformmark.cpp

```diff
@@ -6,21 +6,26 @@
 #include "control/controlobject.h"
 #include "control/controlproxy.h"
 #include "widget/wskincolor.h"
+#include "util/memory.h"
 
 WaveformMark::WaveformMark()
-    : m_pPointCos(nullptr) {
+    : m_iIndex(-1) {
 }
 
-WaveformMark::~WaveformMark() {
-    delete m_pPointCos;
+WaveformMark::WaveformMark(const WaveformMark* p_waveformMark)
```

Comment:
Why pass WaveformMark by pointer?

Passing by const-ref would make this constructor a copy-constructor. You are explicitly copying all members except m_pPointCos, m_iIndex, and m_image. Why is m_iIndex left uninitialized? Why is m_image not copied?

Only copying the other object partially is a sign that the members of this class don't belong together, at least not at the same level of abstraction. What about moving the copyable part of this class into a separate class with a default copy constructor?